File: 백준/Bronze/1157. 단어 공부/단어 공부.cc
#include <iostream>

using namespace std;

string s;
int arr[26];
int cnt = 0;
int num;
char ans;

int main() {
  cin.tie(0) -> ios::sync_with_stdio(0);

  cin >> s;

  for(auto it = s.begin(); it != s.end(); it++) {
    num = toupper(*it) - 65;
    arr[num]++;
    if(arr[num] > cnt) {
      cnt = arr[num];
      ans = toupper(*it);
    } else if(arr[num] == cnt) {
      ans = '?';
    }
  }

  cout << ans << "\n";

  return 0;
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.py
word = input()
arr = [0 for _ in range(26)]

for c in word:
    arr[ord(str.lower(c)) - 97]+= 1

ans = '?'
cnt = 0
dup = False
for i in range(26):
    if arr[i] == cnt:
        dup = True
    elif arr[i] > cnt:
        ans = chr(i + 65)
        cnt = arr[i]
        dup = False
        
if dup:
    ans = '?'
    
print(ans)